src/org/labkey/test/components/domain/DomainFormPanel.java

```diff
@@ -120,6 +121,15 @@ else if (validator instanceof FieldDefinition.RangeValidator)
                 {
                     rangeValidators.add((FieldDefinition.RangeValidator) validator);
                 }
+                else if (validator instanceof FieldDefinition.TextChoiceValidator)
+                {
+                    // TextChoice is a field type and not an attribute on a field, any other validators that are applied
+                    // will be ignored (TextChoice field cannot have validators).
+                    textChoiceValidator = (FieldDefinition.TextChoiceValidator) validator;
+                    regexValidators = new ArrayList<>();
+                    rangeValidators = new ArrayList<>();
+                    break;
+                }
```

Comment:
Should just error if there are multiple validators instead of ignoring them silently.
```suggestion
                else if (validator instanceof FieldDefinition.TextChoiceValidator textChoiceValidator)
                {
                    // TextChoice is a field type; implemented using a special validator. TextChoice field cannot have other validators.
                    if (validators.size() > 1)
                    {
                        throw new IllegalArgumentException("TextChoice fields cannot have additional validators.");
                    }
                    fieldRow.setTextChoiceValues(textChoiceValidator.getValues());
                }
```

Reply:
Fair enough. Suggested change applied.